### PR TITLE
add SOA example

### LINKED ISSUE #2985: feat: service oriented example with V10
### Describe the feature you'd like to request

I'm looking for an example of how to implement service-oriented / microservices approach with tRPC V10.

With V9 we have working solution based on finding from https://github.com/trpc/trpc/issues/2418 and https://github.com/trpc/trpc/pull/1226a.

### Describe the solution you'd like to see

Could you update https://github.com/trpc/trpc/pull/1226 to use V10

### Desribe alternate solutions

Provide example how to implement `CombinedServers` approach with `createTRPCProxyClient`

### Additional information

_No response_

### 👨‍👧‍👦 Contributing

- [ ] 🙋‍♂️ Yes, I'd be down to file a PR implementing this feature!

### PR DESCRIPTION
**Pull request by KATT**
Closes #2985

## 🎯 Changes

- SOA example
- The "gateway" doesn't actually create a server; the _client_ decides at runtime which service to call depending on the prefix of the call
- Caveat: think all services need the same context to be merged

## ✅ Checklist

- [ ] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [ ] If necessary, I have added documentation related to the changes made.
- [ ] I have added or updated the tests related to the changes made.

### CODE REVIEW

**Comment by miekassu** [comment]
Thanks for the updated example @KATT, this helped us forward.

Based on your example and findings from https://github.com/trpc/trpc/pull/2165 I did an example repo with lambda adapters https://github.com/miekassu/trpc-aws-lambda

Maybe this can be added in examples https://github.com/trpc/trpc/issues/2152?

**Comment by chrisabrams** [comment]
Hi @KATT overall this direction makes sense. I currently am using v10 and have a need for multiple services. I've been wondering how to manage multiple different services. I didn't like the idea of making a proxy TRPC server that would then turn around and go to other TRPC servers. Having a namespace on the client seems like a simpler solution. If you need the server-based approach then you'd have to do something custom anyway. Making each server have the same context doesn't seem like a big deal; it would be confusing if each service behaved differently.

**Comment by KATT** [comment]
If this behaviour would break it would be considered a breaking change that should've been a major. Open an issue if we ever accidentally break it

**Comment by KATT** [comment]
That said, I would love to see a POC of your CF setup! I can be swayed :)

File: examples/.test/big-router-declaration/package.json [hunk 1]
```diff
@@ -22,6 +22,7 @@
     "@types/node": "^18.7.20",
     "@types/react": "^18.0.9",
     "@types/react-dom": "^18.0.5",
+    "node-fetch": "^2.6.1",
```
Comment (KATT) [inline examples/.test/big-router-declaration/package.json:25]:
```suggestion
```

**Comment by KATT** [comment]
@chrisabrams, @miekassu, @Eduard-Hasa 

If you're using SOA, I bet you work for fairly large organizations on business-critical applications, it would be great if you could convince your employers to sponsor tRPC - https://trpc.io/docs/sponsor

**Comment by Eduard-Hasa** [comment]
@KATT I would most definitely push for it and have been raving about trpc to everyone in my circle. Im, unfortunately, using this only for my own side project in my free time. The reason I asked for this feature is because I dont want to limit my api to a single cloudflare worker for reasons that may or may not be justified by some. But I have asked in the cloudflare discord and many have told me to probably keep the api separate for a cleaner and efficient workflow. Cloudflare workers also have limits, so trying to cram everything in one would be counter intuitive. By the way im going with cloudflare workers instead of vercel edge api because of Cloudflare's Durable Objects. They are amazing for real time applications, which is what im building. Nothing out there competes imo and I think trpc the Cloudflare platform is a wise decision. I will put together my sample project to add to your documentation. Im really happy with it.

Im currently leading a software engineering project at General Motors and will definitely rave about trpc to everyone I know though. After a few weeks of developing a poc on my side project, im now well informed on trpc and will definitely push it where I can.

**Comment by KATT** [comment]
Ooops, thanks for the nudge, I fixed the link! Also thanks so much for your kind words ❤️ 

Would love to see a demo of your CF setup :)

**Pull request merged**